cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/JsonPaginatedListResponder.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.gateway.handlers;
+
+import com.google.gson.Gson;
+import io.cdap.http.HttpResponder;
+
+import java.io.IOException;
+import java.util.function.Function;
+
+/**
+ * JsonPaginatedListResponder - for streaming applications using Pagination
```

Comment:
Add better description. E.g. how is it good for pagination, what does it do, how response looks like